app/src/main/java/android/coding/interview/makeitawesome/fragment/PicturesFragment.java

```diff
@@ -25,13 +48,87 @@ public static Fragment newInstance() {
+        setupRecyclerView(mRecyclerView);
+        return mRecyclerView;
     }
 
     private void setupRecyclerView(RecyclerView recyclerView) {
+        mRecyclerView.addItemDecoration(new DividerItemDecoration(recyclerView.getContext(), DividerItemDecoration.VERTICAL_LIST));
+        mRecyclerView.setAdapter(mPhotosAdapter);
+        mPhotosAdapter.setClickListener(this);
         recyclerView.setLayoutManager(new LinearLayoutManager(recyclerView.getContext()));
-        recyclerView.setAdapter(new PhotosAdapter());
+    }
+
+    @Override
+    public void itemClicked(View view, int position) {
+        Intent intent = new Intent(getActivity(), Picture.class);
+        String url = mData.get(position).getUrl();
+
+        intent.putExtra(Picture.IMAGEURL, url);
+        startActivity(intent);
+    }
+
+    /**
+     * download the data
+     */
+    private class HttpGetTask extends AsyncTask<Void, Void, ArrayList<ImageData>> {
+        AndroidHttpClient mClient = AndroidHttpClient.newInstance("");
+
+
+        @Override
+        protected ArrayList<ImageData> doInBackground(Void... unused) {
+            Log.v(TAG, "doInBackground");
+
+            HttpGet request = new HttpGet(URL);
+            JSONResponseHandler searchresponseHandler = new JSONResponseHandler();
+
+            try {
+                return mClient.execute(request, searchresponseHandler);
+            } catch (ClientProtocolException exception) {
+                exception.printStackTrace();
+            } catch (IOException ioexception) {
+                ioexception.printStackTrace();
+            } finally {
+                if (mClient != null) mClient.close();
+            }
+            return null;
+        }
+
+        @Override
+        protected void onPostExecute(ArrayList<ImageData> result) {
+            Log.v(TAG, "onPostExecute");
+            ArrayList<ImageData> searchResults = new ArrayList<>();
+
+            if (mClient != null && result != null) {
+                Log.v(TAG, "search complete");
+                /*for (Map.Entry<String, ImageData> mLibraryItem : result.getphotos().entrySet()) {
+                    searchResults.add(mLibraryItem.getValue());
+                }*/
+                //for (ImageData data : )
+                for (ImageData mData : result){
+                    searchResults.add(mData);
+                }
+            }
+            if ((result == null || result.size() == 0)) {
+                //todo: if there's no results show no results found
+                //tv_no_results.setVisibility(View.VISIBLE);
+            } else {
+                //tv_no_results.setVisibility(View.GONE);
+                int counter = 0;
+                for (ImageData imageData: searchResults) {
+                    mPhotosAdapter.insertmData(imageData, counter);
+                    counter++;
+                }
+            }
+        }
```

Comment:
Asynctask and not using any cool network library makes me think he is not super expert in Android nor in Java. But he doesn't code like a newbie.